src/lib.rs

```diff
@@ -93,6 +99,50 @@ fn notify_retry(e: &Error, duration: Duration) {
     }
 }
 
+/// Creates a SHA-256 digest of the given bytes encoded as a hex string.
+fn sha256_hex_string(bytes: impl AsRef<[u8]>) -> String {
+    let mut hash = Sha256::new();
+    hash.update(bytes);
+    hex::encode(hash.finalize())
+}
+
+trait DateTimeExt {
+    /// Converts a [`DateTime`] to a HTTP date string.
+    ///
+    /// See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Date
+    fn to_http_date(&self) -> String;
```

Comment:
This was implemented to replace the use of `to_rfc2822`; turns out that Azure Storage actually validates the `Date` header for signed requests and didn't like the incorrect use of the RFC 2822 format.